jsonschema/tests/test_cli.py

```diff
@@ -683,6 +683,23 @@ def test_successful_validation__of_just_the_schema_pretty_output(self):
             stderr="",
         )
 
+    def test_successful_validation_with_specifying_base_uri(self):
+        schema = """\
+                {"type": "object", "properties": {"KEY1":
+                {"$ref": "schema.json#definitions/schemas"}},
+                "definitions": {"schemas": {"type": "string"}}}
+                """
+        fp = open("schema.json", "w+")
```

Comment:
Can we use `tempfile.NamedTemporaryFile` here, bit clearer, and doesn't depend on the current directory.

Also, as-is this would leak a file onto the disk if the test failed somewhere in the middle, another thing the above should fix.

Reply:
Can we add a schema.json file in the tests directory like tests/schema.json, or in other directory, because I found that if `tempfile.NamedTemporaryFile` is not closed in windows, there will be permission problems if it is opened again, 
so I have to close it before using, at this time, the file will disappear. If the  `delete` option  is set to `False`, the generated temporary files still need to be deleted manually.

Reply:
You can do those via:

```
schema_file = tempfile.NamedTemporaryFile(delete=False)
self.addCleanup(os.remove, schema_file.name)
```

or thereabouts.